src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -36,6 +55,30 @@ class Ascendance extends Analyzer {
   }
 
   on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    const target = this.enemies.getEntity(event);
+
+
+    if(spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) {
+      if(target && target.getBuffUptime(SPELLS.FLAME_SHOCK.id) < ASCENDENCE_DURATION){
+        this.badFSAscendence++;
+      }
+    }
+
+
+    const ability = this.abilities.getAbility(spellId);
+    if(!ability){
+      return;
+    }
+    if(!this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp) || spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id){
+      return;
+    }
+
+    const gcd = this._resolveAbilityGcdField(ability.gcd);
+    if(!gcd){
+      return;
+    }
+
     if (this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp)) {
```

Comment:
This if statement will always be true since it's the opposite of the first part of the one in line 73